simpleclient_spring_web/src/main/java/io/prometheus/client/spring/web/PrometheusTimeMethods.java

```diff
@@ -0,0 +1,46 @@
+package io.prometheus.client.spring.web;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Enable Spring-AOP-based automated method timing for the annotated method or all methods in an annotated class.
+ * The timings will be recorded in a Histogram with the default metric name of {@code prometheus_method_timing},
+ * which can be overridden by providing a {@code value} parameter in the annotation, and optionally a {@code help}
+ * parameter and a single label of {@code signature} whose values will be {@code ClassOrInterfaceName.methodName(..)}.
+ *
+ * To properly work, {@link EnablePrometheusTiming} must be specified somewhere in your application configuration.
+ *
+ *  <pre><code>
+ * {@literal @}PrometheusTimeMethods("my_app_timer_seconds")
```

Comment:
You know, somehow it hadn't really occurred to me that class methods would be calling each other in the same class, within loops, recursively, etc. I'm too used to languages like Go or JavaScript where methods tend to call functions not associated with an instance.

So yeah, per-method granularity makes a lot more sense.

That should make things a lot more performant too, since as you mentioned, we can cache the child and avoid a lookup.